extensions/wikia/UserActivity/SpecialUserActivityController.class.php

```diff
@@ -0,0 +1,97 @@
+<?php
+
+namespace UserActivity;
+
+class SpecialController extends \WikiaSpecialPageController {
+
+	const DEFAULT_TEMPLATE_ENGINE = \WikiaResponse::TEMPLATE_ENGINE_MUSTACHE;
+	const PAGE_NAME = 'UserActivity';
+	const PAGE_TITLE = 'User Activity';
+
+	const ITEMS_PER_PAGE = 10;
+
+	public function __construct() {
+		parent::__construct( self::PAGE_NAME, '', $listed = false );
+	}
+
+	/**
+	 * Run before index. First make sure the user can access page, then set the title, disable
+	 * redirects, and add the css.
+	 */
+	public function init() {
+		$this->disableRedirects();
+		$this->setTitle();
+		$this->addCss();
+	}
+
+	/**
+	 * Disable redirects. Some operations called in the controllers will cause the page to
+	 * redirect (eg Article::newFromTitle()). We always want to return to Special:SendEmail.
+	 */
+	private function disableRedirects() {
+		$this->wg->Out->enableRedirects( false );
+	}
+
+	/**
+	 * Set's the <h1> and <title> tags, and suppresses the generic "Special Page" subtitle.
+	 */
+	private function setTitle() {
+		$this->getContext()->getOutput()->setHTMLTitle( self::PAGE_TITLE );
+		$this->getContext()->getOutput()->setPageTitle( self::PAGE_TITLE );
+		$this->wg->SupressPageSubtitle = true;
+	}
+
+	/**
+	 * Adds stylesheet to output
+	 */
+	private function addCss() {
+		$this->response->addAsset( 'special_user_activity_css' );
+	}
+
+	/**
+	 * @template specialUserActivity
+	 */
+	public function index() {
+		if ( $this->wg->User->isAnon() ) {
+			$this->getResponse()->redirect( '/' );
+			return;
+		}
+
+		$limit = $this->getVal( 'limit', self::ITEMS_PER_PAGE );
+		$page = $this->getVal( 'page', 1 );
+		$order = $this->getVal( 'order', 'lastedit:desc' );
+
+		$resp = $this->sendRequest( 'UserActivity\Controller', 'index', [
+			'limit' => $limit,
+			'offset' => ( $page - 1 ) * $limit,
+			'order' => $order,
+		] );
+		$data = $resp->getData();
+
+		$this->getResponse()->setData( [
+			'pageDescription' => wfMessage( 'user-activity-page-description' )->text(),
+			'tableTitle' => wfMessage( 'user-activity-table-title' )->text(),
+			'tableEdits' => wfMessage( 'user-activity-table-edits' )->text(),
+			'tableLastEdit' => wfMessage( 'user-activity-table-lastedit' )->text(),
+			'tableRights' => wfMessage( 'user-activity-table-rights' )->text(),
```

Comment:
Should all of these be `escaped()`? In case some admin decides to override those messages.